File: Projeto em Java/src/controller/LeitorCSV.java
package controller;

import model.Partida;
import model.Cartao;
import model.Gol;
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class LeitorCSV {
    
    public List<Partida> carregarPartidas(String caminho) {
        List<Partida> partidas = new ArrayList<>();
        try (BufferedReader br = new BufferedReader(new FileReader(caminho))){
            String linha = br.readLine(); //pula o cabecalho
            while ((linha = br.readLine()) != null){
                String[] colunas = linha.split(",");
                if (colunas.length < 6){ //tratamento para linhas quebradas
                    continue;
                }
                try {
                    String mandante = colunas[0].trim().replace("\"", "");
                    String visitante = colunas[1].trim().replace("\"", "");
                    String vencedor = colunas[2].trim().replace("\"", "");
                    int placarMandante = Integer.parseInt(colunas[3].trim().replace("\"", ""));
                    int placarVisitante = Integer.parseInt(colunas[4].trim().replace("\"", ""));
                    String vencedorEstado = colunas[5].trim().replace("\"", "");

                    if (vencedorEstado.isEmpty()){ //tratamento para caso de empate
                        vencedorEstado = null;
                    }

                    partidas.add(new Partida(mandante, visitante, vencedor, placarMandante, placarVisitante, vencedorEstado));
                }catch (NumberFormatException e){
                    System.err.println("Erro ao converter placar para número na linha: " + linha);
                }
            }
        }catch (FileNotFoundException e){
            System.err.println("Arquivo 'campeonato-brasileiro-full.csv' nao encontrado no caminho: " + caminho);
        }catch (IOException e){
            System.err.println("Erro de leitura do arquivo: " + e.getMessage());
        }
        return partidas;
    }

    public List<Gol> carregarGols(String caminho){
        List<Gol> gols = new ArrayList<>();
        try (BufferedReader br = new BufferedReader(new FileReader(caminho))){
            String linha = br.readLine(); //pula o cabecalho
            while ((linha = br.readLine()) != null){
                String[] colunas = linha.split(",");
                if (colunas.length < 3){ //tratamento para caso de linha quebrada
                    continue;
                }

                String clube = colunas[0].trim().replace("\"", "");
                String atleta = colunas[1].trim().replace("\"", "");
                String tipo = colunas[2].trim().replace("\"", "");

                gols.add(new Gol(clube, atleta, tipo));
            }
        }catch (FileNotFoundException e){
            System.err.println("Arquivo 'campeonato-brasileiro-gols.csv' nao encontrado no caminho: " + caminho);
        }catch (IOException e){
            System.err.println("Erro de leitura do arquivo: " + e.getMessage());
        }
        return gols;
    }

    public List<Cartao> carregarCartoes(String caminho){
        List<Cartao> cartoes = new ArrayList<>();
        try (BufferedReader br = new BufferedReader(new FileReader(caminho))){
            String linha = br.readLine(); //pula o cabecalho
            while ((linha = br.readLine()) != null){
                String[] colunas = linha.split(",");
                if (colunas.length < 3){ //tratamento para caso de linha quebrada
                    continue;
                }

                String clube = colunas[0].trim().replace("\"", "");
                String tipo = colunas[1].trim().replace("\"", "");
                String atleta = colunas[2].trim().replace("\"", "");

                cartoes.add(new Cartao(clube, tipo, atleta));
            }
        }catch (FileNotFoundException e){
            System.err.println("Arquivo 'campeonato-brasileiro-cartoes.csv' nao encontrado no caminho: " + caminho);
        }catch (IOException e){
            System.err.println("Erro de leitura no arquivo: " + e.getMessage());
        }
        return cartoes;
    }
}

File: Projeto em Python/controller/leitor_csv.py
import csv
from model.partida import Partida
from model.gol import Gol
from model.cartao import Cartao

class LeitorCSV:
    def carregar_partidas(self, caminho):
        partidas = []
        try:
            with open(caminho, 'r', encoding='utf-8') as arquivo:
                leitor = csv.reader(arquivo)
                next(leitor) #pulando o cabeçalho
                for linha in leitor:
                    if len(linha) < 6: #pulando casos de linhas com dados ausentes
                        continue
                    try:
                        partida = Partida(
                            mandante=linha[0].strip().replace('"', ''),
                            visitante=linha[1].strip().replace('"', ''),
                            vencedor=linha[2].strip().replace('"', ''),
                            placar_mandante=int(linha[3].strip().replace('"', '')),
                            placar_visitante=int(linha[4].strip().replace('"', '')),
                            vencedor_estado=linha[5].strip().replace('"', '') if linha[5].strip().replace('"', '') else None
                        )
                        partidas.append(partida)
                    except (ValueError, IndexError) as e:
                        print(f"Erro ao processar a linha: {linha}. Erro: {e}")
        except FileNotFoundError:
            print(f"Arquivo não encontrado: {caminho}")
        return partidas
    
    def carregar_gols(self, caminho):
        gols = []
        try:
            with open(caminho, 'r', encoding='utf-8') as arquivo:
                leitor = csv.reader(arquivo)
                next(leitor) #pulando o cabeçalho
                for linha in leitor:
                    if len(linha) < 3: #pulando casos de linhas com dados ausentes
                        continue
                    gol = Gol(
                        clube=linha[0].strip().replace('"', ''),
                        atleta=linha[1].strip().replace('"', ''),
                        tipo=linha[2].strip().replace('"', '')
                    )
                    gols.append(gol)
        except FileNotFoundError:
            print(f"Arquivo não encontrado: {caminho}")
        return gols
    
    def carregar_cartoes(self, caminho):
        cartoes = []
        try:
            with open(caminho, 'r', encoding='utf-8') as arquivo:
                leitor = csv.reader(arquivo)
                next(leitor) #pulando o cabeçalho
                for linha in leitor:
                    if len(linha) < 3: #pulando casos de linhas com dados ausentes
                        continue
                    cartao = Cartao(
                        clube=linha[0].strip().replace('"', ''),
                        tipo=linha[1].strip().replace('"', ''),
                        atleta=linha[2].strip().replace('"', '')
                    )
                    cartoes.append(cartao)
        except FileNotFoundError:
            print(f"Arquivo não encontrado: {caminho}")
        return cartoes
